tests/test_butler.py

```diff
@@ -619,6 +619,13 @@ def setUp(self):
         uri = ButlerURI(config[".datastore.datastore.root"])
         self.bucketName = uri.netloc
 
+        # set up some fake credentials if they do not exist
+        if not os.path.exists("~/.aws/credentials"):
+            if "AWS_ACCESS_KEY_ID" not in os.environ:
```

Comment:
Similarly here? You want to allow people to use their own credentials? Meaning that somehow these tests can run without a mock?